packages/shopify-app-session-storage-redis/src/redis.ts

```diff
@@ -151,5 +153,42 @@ export class RedisSessionStorage implements SessionStorage {
       this.client.on('error', this.options.onError);
     }
     await this.client.connect();
+    await this.migrate();
+  }
+
+  private async migrate() {
+    const migrationsRecord = await this.client.get(this.fullKey('migrations'));
+    if (migrationsRecord) {
+      // need to check if we need to migrate
+      // as of 2.0.0, this code is illustrative only!
+      const migrations = JSON.parse(migrationsRecord);
+      if (!migrations.migrateToVersion2_0_0) {
```

Comment:
There seems to be a lot of common code between the 2 branches of your `if`?Can they be mutualized ?

Reply:
Yes! Now rewritten.